pkg/cfn/builder/vpc.go

```diff
@@ -10,9 +11,9 @@ import (
 	"github.com/weaveworks/eksctl/pkg/vpc"
 )
 
-func (c *ClusterResourceSet) addSubnets(refRT *gfn.Value, topology api.SubnetTopology, subnets map[string]api.Network) {
+func (c *ClusterResourceSet) addSubnets(refRT *gfn.Value, topology api.SubnetTopology, subnets map[string]api.Network, kind string) {
 	for az, subnet := range subnets {
-		alias := string(topology) + strings.ToUpper(strings.Join(strings.Split(az, "-"), ""))
+		alias := string(kind) + string(topology) + strings.ToUpper(strings.Join(strings.Split(az, "-"), ""))
```

Comment:
We will have to maintain names of existing keys as they are, so please only add prefix for new subnets... That is because logical names inside the stack need to remain the same for our append-only stack update code to function as expect, but also in general that is essential in CloudFormation.